lib/health_cards/qbp_client.rb

```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: true
+
+require 'savon'
+require 'ruby-hl7'
+require 'faraday'
+require_relative 'qpd'
+
+module HealthCards
+  # Send, receive, and translate HL7 V2 messages from the QBP client of the IIS sandbox
+  module QBPClient # rubocop:disable Metrics/ModuleLength # Disabling since module will be ported and refactored
+    extend self
+    # Query a patient's Immunization history from the IIS Sandbox
+    # @param patient_info [Hash] Patient demographic info sent from the IIS Consumer Portal
+    # @param credentials [Hash] (optional) User Id, Password, and Facility Id for the IIS Sandbox login
+    # @return [HL7::Message] The HL7 V2 Response message from the IIS-Sandbox
+    def query(patient_info,
+              sandbox_credentials = { username: Rails.application.config.username,
+                                      password: Rails.application.config.password,
+                                      facilityID: Rails.application.config.facilityID })
+      raise HealthCards::InvalidSandboxCredentialsError unless valid_credentials?(sandbox_credentials)
+
+      service_def = 'lib/assets/service.wsdl'
+      client = Savon.client(wsdl: service_def,
+                            endpoint: 'http://vci.mitre.org:8081/iis-sandbox/soap',
+                            pretty_print_xml: true)
+      # Check if client is configured properly
+      raise HealthCards::OperationNotSupportedError unless client.operations.include?(:submit_single_message)
+
+      check_client_connectivity(client) if client.operations.include?(:connectivity_test)
```

Comment:
We don't need to check this with every query, just on-demand.

Reply:
Good point, I'll just leave it as a function that for now is called by just the one test case.